### PR TITLE
add 10-gumball-machine-front-end example

### PR DESCRIPTION
**Pull request by azizi-a**

### CODE REVIEW

File: step-by-step/10-gumball-machine-front-end/README.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+   - [Setup](#setup)
+   - [Creating a Radix Wallet Stokenet Account](#creating-a-radix-wallet-stokenet-account)
+   - [Scrypto](#scrypto)
+      - [Build the scrypto package:](#build-the-scrypto-package)
+      - [Deploy the package to Stokenet:](#deploy-the-package-to-stokenet)
+   - [Creating a dApp Definition](#creating-a-dapp-definition)
+   - [Front End Client](#front-end-client)
+   - [Using the dApp](#using-the-dapp)
+
+We have a deployed a Gumball Machine package to the test network in example 8,
```
Comment (motosharpley) [inline step-by-step/10-gumball-machine-front-end/README.md:24]: We have a deployed Gumball Machine

File: step-by-step/10-gumball-machine-front-end/README.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+;
+CALL_METHOD
+  Address("${componentAddress}")
+  "refill_gumball_machine"
+;
+CALL_METHOD
+  Address("${accountAddress}")
+  "deposit_batch"
+  Expression("ENTIRE_WORKTOP")
+;`;
```
Comment (motosharpley) [inline step-by-step/10-gumball-machine-front-end/README.md:82]: ; on a new Line I thought it was extra at first glance

File: step-by-step/10-gumball-machine-front-end/README.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        await rdt.gatewayApi.state.getEntityDetailsVaultAggregated(
+          componentAddress
+        );
+      console.log("Component Details:", componentDetails);
+
+      // Get the price, number of gumballs, and earnings from the component state
+      const price = componentDetails.details.state.fields.find(
+        (field) => field.field_name === "price"
+      )?.value;
+      const noOfGumballs = componentDetails.fungible_resources.items.find(
```
Comment (motosharpley) [inline step-by-step/10-gumball-machine-front-end/README.md:159]: it's a little thing but how about `numOfGumballs` vs `noOfGumballs`

File: step-by-step/10-gumball-machine-front-end/README.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+1. The Radix Wallet
+   [more info here](https://docs.radixdlt.com/docs/radix-wallet-overview)
+2. The Radix Wallet Connector extension. Download from the
+   [chrome store](https://chromewebstore.google.com/detail/radix-wallet-connector/bfeplaecgkoeckiidkgkmlllfbaeplgm)
+   or [download from github](https://github.com/radixdlt/connector-extension/)
+
+### Setup
+
+There are separate setup steps for the scrypto package and the front end client.
+You'll also need to Stokenet account in your Radix Wallet, with funds.
```
Comment (motosharpley) [inline step-by-step/10-gumball-machine-front-end/README.md:195]: You'll also need a Stokenet account in your Radix Wallet, with XRD to pay fees.

File: step-by-step/10-gumball-machine-front-end/README.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+
+#### Creating a Radix Wallet Stokenet Account
+
+If you already have a Stokenet account with funds you can skip this step.
+
+1. If you haven't already, open the Radix Wallet and on your phone and run
```
Comment (motosharpley) [inline step-by-step/10-gumball-machine-front-end/README.md:201]: open the Radix Wallet on your phone and run

File: step-by-step/10-gumball-machine-front-end/README.md (hunk 2 above)
Comment (azizi-a) [reply step-by-step/10-gumball-machine-front-end/README.md:82]: My formatter keeps putting it back in the same place 😆

**Review by motosharpley — APPROVED** [review]

**Pull request merged**